### PR TITLE
Add a test for lowercasing a capital I with a dot above

### PR DESCRIPTION
**Pull request by nox**
[Review on Reviewable](https://reviewable.io/reviews/w3c/csswg-test/828)

### CODE REVIEW

**Comment by r12a** [comment]
i believe the ref file for this test is incorrect, since the combining character is inappropriate. The default lowercase of U+0130   LATIN CAPITAL LETTER I WITH DOT ABOVE is simply U+0069   LATIN SMALL LETTER I according to the Unicode Standard.

btw, this test already exists in a large batch that is currently awaiting resolution of a pull request. See https://github.com/w3c/csswg-test/pull/763

hope that helps

(btw, i do wish that tests would include assertions.  It saves a lot of time for review if you know what you should be looking for. Eg. i'm assuming that this test is checking the default case folding per Unicode, rather than the tailored case conversion used for Turkish - in which case, the test would be incorrectly set up)

**Comment by nox** [comment]
@r12a No, the *default* lowercase for &lt;U+0130 LATIN CAPITAL LETTER I WITH DOT ABOVE> is &lt;U+0069 LATIN SMALL LETTER I, U+0307 COMBINING DOT ABOVE>. It is just U+0069 LATIN SMALL LETTER I in the Turkish locale.

See [CaseFolding.txt](http://www.unicode.org/Public/8.0.0/ucd/CaseFolding.txt):

```
#
# ================================================================================
# Format
# ================================================================================
# The entries in this file are in the following machine-readable format:
#
# <code>; <status>; <mapping>; # <name>
#
# The status field is:
# C: common case folding, common mappings shared by both simple and full mappings.
# F: full case folding, mappings that cause strings to grow in length. Multiple characters are separated by spaces.
# S: simple case folding, mappings to single characters where different from F.
# T: special case for uppercase I and dotted uppercase I
#    - For non-Turkic languages, this mapping is normally not used.
#    - For Turkic languages (tr, az), this mapping can be used instead of the normal mapping for these characters.
#      Note that the Turkic mappings do not maintain canonical equivalence without additional processing.
#      See the discussions of case mapping in the Unicode Standard for more information.
#
# Usage:
#  A. To do a simple case folding, use the mappings with status C + S.
#  B. To do a full case folding, use the mappings with status C + F.
#
#    The mappings with status T can be used or omitted depending on the desired case-folding
#    behavior. (The default option is to exclude them.)
```

```
0130; F; 0069 0307; # LATIN CAPITAL LETTER I WITH DOT ABOVE
0130; T; 0069; # LATIN CAPITAL LETTER I WITH DOT ABOVE
```

That's why the test includes  `lang=en` explicitly.

**Comment by nox** [comment]
> (btw, i do wish that tests would include assertions. It saves a lot of time for review if you know what you should be looking for. Eg. i'm assuming that this test is checking the default case folding per Unicode, rather than the tailored case conversion used for Turkish - in which case, the test would be incorrectly set up)

I have no idea what you mean by assertions. This test is pretty much correct however you look at it.

**Comment by plinss** [comment]
@nox See http://testthewebforward.org/docs/css-metadata.html#test-assertions 

It's metadata explaining what the test is trying to prove in plain english. It's very useful for reviewers to help understand the test.

**Comment by nox** [comment]
I added one.

**Comment by Ms2ger** [comment]
Rebase please.

**Comment by nox** [comment]
Not sure why, but done.

**Comment by kojiishi** [comment]
I think this is the correct expected result unless the text is tagged as Turkic. @r12a, do you agree?

For the directory and the spec link, however, [text-transform in CSS 2](http://www.w3.org/TR/CSS21/text.html#caps-prop) does not require full mapping, only [CSS Text Level 3](https://drafts.csswg.org/css-text-3/#text-transform-property) does, so I think it's more appropriate to have this test in css-text-3 directory and the spec link points to CSS Text Level 3.

[css-text-3/text-transform/text-transform-upperlower-006.html](https://github.com/w3c/csswg-test/blob/master/css-text-3/text-transform/text-transform-upperlower-006.html) includes U+0130 but the expected result seems to assume it's in Turkic languages. With @r12a's confirmation, it'd be great if you could fix it.

A minor point but lang tag could affect font choices, so when you have it in the test, it's recommended to have it in ref too.

**Comment by nox** [comment]
@kojiishi Fixed both tests.

**Comment by kojiishi** [comment]
Reviewed 5 of 5 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/csswg-test/828)*
